mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td

```diff
@@ -153,14 +153,23 @@ class NVVM_IntrOp<string mnem, list<Trait> traits = [],
 // NVVM special register op definitions
 //===----------------------------------------------------------------------===//
 
-class NVVM_SpecialRegisterOp<string mnemonic, list<Trait> traits = []> :
+// NVVM_PureSpecialRegisterOp represents special register ops that can
+// speculated and does not touch memory. These operations are always
+// legal to hoist or sink.
```

Comment:
I don't think we need extra comment here. Pure is self-explanatory

Reply:
Got it. Reverted the comment